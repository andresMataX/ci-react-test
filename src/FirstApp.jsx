// En Vite debe: npm install prop-types
import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h1>{title}</h1>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>Soy un subtítulo con el poto de Moka {subtitle + 0} </p>
    </>
  )
}

FirstApp.propTypes = {
  // Definimos el tipo de dato de las props y si es obligatorio al usar el componente
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
}

// Entran antes de los props, por eso no marcan error
FirstApp.defaultProps = {
  // title: 'No hay título',
  subtitle: 'No hay subtítulo',
  name: 'Andrés',
}
