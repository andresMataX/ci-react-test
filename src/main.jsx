import React from 'react';
import ReactDOM from 'react-dom/client';

import { CounterApp } from './CounterApp'

// Estilos globales de la aplicaci√≥n
import './main.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={0} />
    </React.StrictMode>
)