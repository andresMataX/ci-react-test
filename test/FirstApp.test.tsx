/** @jest-environment jsdom */
import React from 'react'
import { describe, test } from '@jest/globals'
import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Testing in <FirstApp.test />', () => {
  test('debe de hacer match con el snapshot', () => {
    const title = 'Hola, pepe'

    render(<FirstApp title={title} subtitle={12} />)
  })
})
