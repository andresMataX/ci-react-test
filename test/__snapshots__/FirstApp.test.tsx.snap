// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <FirstApp.test /> should hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, pepe
     
  </h1>
  <p>
    Soy un subtítulo con el poto de Moka 
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
</div>
`;
